### DISCUSSION TITLE
ModelJoins struct

### DISCUSSION THREAD

**Discussion by exseniorastronaut** [aarondl/sqlboiler#888] [Ideas]
It would be nice if the framework was generating a ModelJoins struct for example:
```go
var CourseJoins = struct {
	University string
}{
	University: "universities on courses.university_id = universities.id",
}
```

and then

```go
Courses(
    InnerJoin(CourseJoins.University),
    UniversityWhere.Name.EQ("test"),
).All(ctx, exec)

**Comment by aarondl** [comment]
I think we really need to try to rethink the way inner joins are done. They're extremely noisy. This kind of  covers up the noise but we can probably do better.
